torch/_inductor/ir.py

```diff
@@ -1252,8 +1253,21 @@ def _dynamic_reshape_indexer(old_size, new_size):
         size_hint = V.graph.sizevars.size_hint
         vars = [sympy_symbol(f"view{i}") for i in range(len(new_size))]
 
-        stack_new = list(zip(vars, new_size))
-        stack_old = list(old_size)
+        def any0(l):
+            return any([size_hint(x) for x in list(l)])
+
+        if any0(old_size) or any0(new_size):
+            assert any0(old_size) and any0(new_size)
+            view_expr = [sympy.Integer(0)] * len(old_size)
+            stack_new = []
+            stack_old = []
+            numel = math.prod(vars, start=sympy.Integer(1))
+            # this is wrong as sympy_symbol is always positive
+            V.graph.sizevars.guard_equals(numel, sympy.Integer(0))
```

Comment:
@ngimel this guard is wrong as it's never `True` since `sympy_symbol` only creates positive symbols. I could just use `sympy.Symbol()` c-tor to create integer symbols but I wonder if this might break some invariants in inductor?

Reply:
@ngimel I've also tried to integrate zero-dim handling logic in the loop below, but it looked even worse than this 😢

Reply:
I think the guard should be on `sympy_product` of elements in the `old_size` to be equal to 0, and elements in the `new_size` to be equal to 0, and a special reindexer function in this case should always return 0, because that's what your offset is going to be.